### ISSUE TITLE
Query strings parsed as part of Controller Method in Google AppEngine

### ISSUE THREAD

**Issue by lesmo** [cakephp/cakephp#12838]
* CakePHP Version: 3.6.13
* Platform and Target: Google AppEngine Standard PHP 7.2

### What you did
Visting a URL like this:
```
*.appspot.com/member/account/login?redirect=%2Fmember
```

With routing setup like this:
```php
Router::prefix('member', function (RouteBuilder $routes) {
    $routes->connect('/login', ['controller' => 'Account', 'action' => 'login']);
});
```

```composer.json``` requiring these:
```json
    "require": {
        "php": ">=5.6",
        "cakephp/cakephp": "3.6.*",
        "cakephp/localized": "^3.5",
        "cakephp/migrations": "^2.0.0",
        "cakephp/plugin-installer": "^1.0",
        "jeremyharris/cakephp-lazyload": "^2.1",
        "josegonzalez/dotenv": "3.*",
        "lilhermit/cakephp-plugin-bootstrap4": "4.0.0.4002",
        "maiconpinto/cakephp-adminlte-theme": "^1.0",
        "mobiledetect/mobiledetectlib": "2.*",
        "smsgatewayme/client": "^0.0.1"
    },
    "require-dev": {
        "cakephp/bake": "^1.1",
        "cakephp/cakephp-codesniffer": "^3.0",
        "cakephp/debug_kit": "^3.15.0",
        "phpunit/php-code-coverage": "^5.3",
        "phpunit/phpunit": "^5.7|^6.0",
        "psy/psysh": "@stable"
    },
```

Caching path has been set to ```/tmp``` due to file writing restrictions.

Configuration ```App.baseUrl``` has been set to ```/``` due to ```/index.php/``` being prepended to all URLs and redirects generated by CakePHP.

### What happened
A ```Cake\Controller\Exception\MissingActionException``` is thrown due to CakePHP trying to find ```App\Controller\Member\LoginredirectController``` (which isn't even a valid Controller name):
![image](https://user-images.githubusercontent.com/2228196/50379130-33320f80-0606-11e9-8f85-f710dfa318e7.png)

### What you expected to happen
Run ```App\Controller\Member\AccountController::login()``` method with ```redirect``` properly parsed as a query string, not part of the Controller method name.

### Additional Notes
* In local environment with nginx + PHP 7.2 routes work as expected.
* App hosted on shared hosting environment with PHP 7.2 + Apache works as expected.
* Project Integration Tests pass correctly.

**Comment by markstory** [comment]
If this works locally it sounds like you have a configuration issue to solve in AppEngine. I don't see how this is a problem in CakePHP. Perhaps start with comparing the `$_SERVER` data in each of your environments and looking for where it differs.

**Comment by ADmad** [comment]
This looks like misconfigured URL rewriting, not an issue with the framework.

**Comment by ADmad** [comment]
The values for `PATH_INFO` and `PHP_SELF` you have showed are incorrect and that's what causing the problem. You'll need to manipulate them yourself and set them to values normally set by other webservers.

The app engine docs also shows example for manipulating the various uri/path related environment variables in PHP https://cloud.google.com/appengine/docs/standard/php/config/mod_rewrite.

So there's no bug in CakePHP code that needs to be fixed. You are welcome to submit an update to the manual showing how the env variables need to be properly set for Appengine.

**Comment by ADmad** [comment]
The default [nginx](https://cloud.google.com/appengine/docs/flexible/php/runtime#customizing_nginx) config for URL rewriting in AppEngine is also different than what's required for [CakePHP](https://book.cakephp.org/3.0/en/installation.html#nginx). So just fixing that also might solve your issue and avoid any manipulation of environment variables in PHP.

**Comment by lesmo** [comment]
Hi! Yeah, it's definitely a change on the GAE environment. I deployed an
app (let's calle it Cake A) in the past without much changes in around
October. I submitted this issue because I didn't change anything and
couldn't make a new project (let's call it Cake B) work.

Cake B didn't work until I did the changes I submitted in my previous
comment. Cake A stopped working too.

Just edit the webroot/index.php with the suggested lines and it'll work
again. Just make sure you're not using FileEngine for cache or sessions,
since those are sure to throw exceptions in AppEngine Standard (AppEngine
Flexible allows writting to the file system).

I guess I'll make a fork of the current CakePHP Skeleton App with these
little adjustments for AppeEngine Standard.
